In [3]:
import numpy as np
import pandas as pd

# DataSet Information
### 1. The topic of our prject is biology, sociology, and psychology. Our title is: Human Interaction 
### 2. We obtained the data set from Kaggle: (insert link)
### 3. This data set studies how different factors contribute to an individuals personality. 

##### Brief introduction summarizing dataset.

In [5]:
human_interaction = pd.read_csv("personality_dataset.csv")
human_interaction

,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert
...,...,...,...,...,...,...,...,...
2895,3.0,No,7.0,6.0,No,6.0,6.0,Extrovert
2896,3.0,No,8.0,3.0,No,14.0,9.0,Extrovert
2897,4.0,Yes,1.0,1.0,Yes,4.0,0.0,Introvert
2898,11.0,Yes,1.0,3.0,Yes,2.0,0.0,Introvert


## Basic Data Exploration

In [6]:
print(f'The first 5 rows of the data set:')
human_interaction.head(5)

The first 5 rows of the data set:


,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert


In [7]:
print(f'The last 5 rows of the data set:')
human_interaction.tail(5)

The last 5 rows of the data set:


,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality
2895,3.0,No,7.0,6.0,No,6.0,6.0,Extrovert
2896,3.0,No,8.0,3.0,No,14.0,9.0,Extrovert
2897,4.0,Yes,1.0,1.0,Yes,4.0,0.0,Introvert
2898,11.0,Yes,1.0,3.0,Yes,2.0,0.0,Introvert
2899,3.0,No,6.0,6.0,No,6.0,9.0,Extrovert


In [8]:
print(f'There are {human_interaction.shape} rows and columns:')

There are (2900, 8) rows and columns:


In [9]:
print(f'The column names are:')
human_interaction.columns

The column names are:


Index(['Time_spent_Alone', 'Stage_fear', 'Social_event_attendance',
       'Going_outside', 'Drained_after_socializing', 'Friends_circle_size',
       'Post_frequency', 'Personality'],
      dtype='object')

## Data Type and Structure

In [10]:
print(f'The data types are:')
human_interaction.info()

The data types are:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2900 entries, 0 to 2899
Data columns (total 8 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Time_spent_Alone           2900 non-null   float64
 1   Stage_fear                 2900 non-null   object 
 2   Social_event_attendance    2900 non-null   float64
 3   Going_outside              2900 non-null   float64
 4   Drained_after_socializing  2900 non-null   object 
 5   Friends_circle_size        2900 non-null   float64
 6   Post_frequency             2900 non-null   float64
 7   Personality                2900 non-null   object 
dtypes: float64(5), object(3)
memory usage: 181.4+ KB


In [11]:
human_interaction.select_dtypes(include='number')

,Time_spent_Alone,Social_event_attendance,Going_outside,Friends_circle_size,Post_frequency
0,4.0,4.0,6.0,13.0,5.0
1,9.0,0.0,0.0,0.0,3.0
2,9.0,1.0,2.0,5.0,2.0
3,0.0,6.0,7.0,14.0,8.0
4,3.0,9.0,4.0,8.0,5.0
...,...,...,...,...,...
2895,3.0,7.0,6.0,6.0,6.0
2896,3.0,8.0,3.0,14.0,9.0
2897,4.0,1.0,1.0,4.0,0.0
2898,11.0,1.0,3.0,2.0,0.0


#### Some of the coluns, such as tage_fear and Drained_after_socializing columns should be converted to numerical to test for a correlational analysis so that we may study the personalit type of the individual.

### Descriptive Summary

In [12]:
hi_summary=human_interaction.describe()
print(hi_summary)
hi_mean=human_interaction['Time_spent_Alone'].mean()
print(hi_mean)
hi_median=human_interaction['Time_spent_Alone'].median()
print(hi_median)
hi_range=human_interaction['Time_spent_Alone'].max()+human_interaction['Time_spent_Alone'].min()
print(hi_range)
hi_std=human_interaction['Time_spent_Alone'].std()
print(hi_std)

       Time_spent_Alone  Social_event_attendance  Going_outside  \
count       2900.000000              2900.000000    2900.000000   
mean           4.505816                 3.963354       3.000000   
std            3.441180                 2.872608       2.221597   
min            0.000000                 0.000000       0.000000   
25%            2.000000                 2.000000       1.000000   
50%            4.000000                 3.963354       3.000000   
75%            7.000000                 6.000000       5.000000   
max           11.000000                10.000000       7.000000   

       Friends_circle_size  Post_frequency  
count          2900.000000     2900.000000  
mean              6.268863        3.564727  
std               4.232340        2.893587  
min               0.000000        0.000000  
25%               3.000000        1.000000  
50%               5.000000        3.000000  
75%              10.000000        6.000000  
max              15.000000       10.

The size of the mean in 6.27 and the standard deviation is 4.23. This shows there is variablility within the data. When looking at the amount of friends within each individual, some have 6 and others have 0and others have 15 friends. While the range is wide, it exemplifies the variety that lies within our data. There are absolutley negative values or missing values within the numerical values.

## Missing or Duplicate Data

In [13]:
missing = human_interaction.isnull().sum()
print("Missing values in each column:\n")
print(missing)

Missing values in each column:

Time_spent_Alone             0
Stage_fear                   0
Social_event_attendance      0
Going_outside                0
Drained_after_socializing    0
Friends_circle_size          0
Post_frequency               0
Personality                  0
dtype: int64


In [14]:
duplicates = human_interaction.duplicated().sum()
print(f"\nThere are {duplicates} duplicate rows in the dataset.")


There are 402 duplicate rows in the dataset.


## Data Wrangling

In [15]:
hi_new=human_interaction.copy()
hi_new

,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert
...,...,...,...,...,...,...,...,...
2895,3.0,No,7.0,6.0,No,6.0,6.0,Extrovert
2896,3.0,No,8.0,3.0,No,14.0,9.0,Extrovert
2897,4.0,Yes,1.0,1.0,Yes,4.0,0.0,Introvert
2898,11.0,Yes,1.0,3.0,Yes,2.0,0.0,Introvert


In [16]:
def drained(Time_spent_Alone):
    if pd.isnull(Time_spent_Alone):
        return 'N/A'
    elif Time_spent_Alone < 5:
        return 1
    else:
        return 2

def fright(Friends_circle_size):
    if pd.isnull(Friends_circle_size):
        return 'N/A'
    elif Friends_circle_size <=5:
        return 2
    else:
        return 1
print

<function print(*args, sep=' ', end='\n', file=None, flush=False)>

In [17]:
hi_new['Drained_after_events']=hi_new['Time_spent_Alone'].apply(drained)
hi_new['Stage_fright']=hi_new['Friends_circle_size'].apply(fright)
hi_new

,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality,Drained_after_events,Stage_fright
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert,1,1
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert,2,2
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert,2,2
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert,1,1
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert,1,1
...,...,...,...,...,...,...,...,...,...,...
2895,3.0,No,7.0,6.0,No,6.0,6.0,Extrovert,1,1
2896,3.0,No,8.0,3.0,No,14.0,9.0,Extrovert,1,1
2897,4.0,Yes,1.0,1.0,Yes,4.0,0.0,Introvert,1,2
2898,11.0,Yes,1.0,3.0,Yes,2.0,0.0,Introvert,2,2


In [18]:
hi_new.head(11)

,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality,Drained_after_events,Stage_fright
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert,1,1
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert,2,2
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert,2,2
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert,1,1
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert,1,1
5,1.0,No,7.0,5.0,No,6.0,6.0,Extrovert,1,1
6,4.0,No,9.0,3.0,No,7.0,7.0,Extrovert,1,1
7,2.0,No,8.0,4.0,No,7.0,8.0,Extrovert,1,1
8,10.0,Yes,1.0,3.0,Yes,0.0,3.0,Introvert,2,2
9,0.0,No,8.0,6.0,No,13.0,8.0,Extrovert,1,1


##### Drained after events, our new column, represents the amount of time a person rates their time being spent alone. 5 hours alone is rated as an idividual who is an extrovert or isnt easily drained.

## Filtering the Data

In [19]:
hi_filtered1=hi_new.copy()
hi_filtered1

,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality,Drained_after_events,Stage_fright
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert,1,1
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert,2,2
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert,2,2
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert,1,1
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert,1,1
...,...,...,...,...,...,...,...,...,...,...
2895,3.0,No,7.0,6.0,No,6.0,6.0,Extrovert,1,1
2896,3.0,No,8.0,3.0,No,14.0,9.0,Extrovert,1,1
2897,4.0,Yes,1.0,1.0,Yes,4.0,0.0,Introvert,1,2
2898,11.0,Yes,1.0,3.0,Yes,2.0,0.0,Introvert,2,2


In [20]:
hi_filtered1.drop(columns=['Drained_after_events'], inplace=True, errors='ignore')
print('\n\n Data after dropping Drained after events column:\n\n')
hi_filtered1



 Data after dropping Drained after events column:




,Time_spent_Alone,Stage_fear,Social_event_attendance,Going_outside,Drained_after_socializing,Friends_circle_size,Post_frequency,Personality,Stage_fright
0,4.0,No,4.0,6.0,No,13.0,5.0,Extrovert,1
1,9.0,Yes,0.0,0.0,Yes,0.0,3.0,Introvert,2
2,9.0,Yes,1.0,2.0,Yes,5.0,2.0,Introvert,2
3,0.0,No,6.0,7.0,No,14.0,8.0,Extrovert,1
4,3.0,No,9.0,4.0,No,8.0,5.0,Extrovert,1
...,...,...,...,...,...,...,...,...,...
2895,3.0,No,7.0,6.0,No,6.0,6.0,Extrovert,1
2896,3.0,No,8.0,3.0,No,14.0,9.0,Extrovert,1
2897,4.0,Yes,1.0,1.0,Yes,4.0,0.0,Introvert,2
2898,11.0,Yes,1.0,3.0,Yes,2.0,0.0,Introvert,2


## Unique Values nd Categories

In [21]:
unique_values=hi_new['Stage_fear'].unique()
print(f'Are there any unique values in the Stage fear column? {unique_values}')

Are there any unique values in the Stage fear column? ['No' 'Yes']


In [22]:
unique_values=hi_new['Drained_after_socializing'].unique()
print(f'Are there any unique values in the Drained after socializing column? {unique_values}')

Are there any unique values in the Drained after socializing column? ['No' 'Yes']


In [24]:
unique_values2=hi_new['Personality'].unique()
print(f'Are there any unique values in the Personality column? {unique_values2}')

Are there any unique values in the Personality column? ['Extrovert' 'Introvert']


The categorical variable that seems the most important for grouping is the Personality column while the Drained_after_socialization and Stage_fear columns are better when looking at correlation analysis. 

## Grouping and Aggregation

In [28]:
personality_by_stage_flight=hi_new.groupby('Personality').agg({
    'Time_spent_Alone': 'mean',
    'Friends_circle_size': 'mean',
    'Social_event_attendance': 'mean'
})
personality_by_stage_flight

,Time_spent_Alone,Friends_circle_size,Social_event_attendance
Personality,,,
Extrovert,2.122869,9.095744,5.977850
Introvert,7.027444,3.277465,1.831621


When looking at this data, iti is evident that theres a difference in habit when looking at the differing personality types. Introverts spend less time with people while extroverts have three tis more ofof events and hold longer friendships as well.

## Summary and Conclusion

An interesting thing about this data set is the contrast between introverts and extroverts and the different was the differences in social energy and the balance of time as well. There was a significant difference in time spent in friendship groups and the amount of social events that were attended. When looking at the column Friends_circle_size, the data distribution shows that the post_frequency column is high, showing that just because someone may have high social engageent does not mean they have high social circles. Within our data, there is a small amount of neutral or ambivert categories. This creates questions about wether this was meant to happen on puprose or if individuals taking the test had to pick between the only given options which were Introvert and Extrovert. 